app/assets/javascripts/vocab_sheet.js

```diff
@@ -91,51 +72,91 @@ $(document).ready(function() {
     }
 
     function checkForMaxLength(elem) {
-      notes = elem.val();
+      var notes = elem.val();
+      var maxLength = elem.attr('maxlength');
 
       if (notes.length >= maxLength) {
         elem.addClass('max-length-reached');
 
         setTimeout(function() {
-          textBox.removeClass('max-length-reached');
+          textBoxes.removeClass('max-length-reached');
         }, 1000);
       }
     }
 
     function setCharacterCount(elem) {
-      clearTimeout(typeTimer);
+      var notes = elem.val();
+      var maxLength = elem.attr('maxlength');
 
-      typeTimer = setTimeout(function() {
-        notes = elem.val();
+      elem.siblings('.character-count__wrap')
+        .children('.character-count__count')
+        .text(maxLength - notes.length);
 
-        elem.siblings('.character-count__wrap')
-          .children('.character-count__count')
-          .text(500 - notes.length);
-
-        updateNotes(formAction, signId, notes);
-      }, 100);
-    }
-
-    function updateNotes(action, signId, notes) {
-      $.ajax({
-        url: action,
-        method: 'PUT',
-        data: {
-          sign_id: signId,
-          notes: notes,
-        },
-        headers: {
-          'X-CSRF-Token': $('meta[name="authenticity-token"]').attr('content'),
-        },
-      }).done(function(data) {
-        // console.log("success!", data)
-      }).fail(function(error) {
-        console.error(error.statusText);
-      });
+      action = getFormAction(elem);
+      signId = getSignIdFromAction(action);
```

Comment:
and here, i think it's a global variable for realz - if it is, can it be in upper case to make it clearer.

Reply:
oh nah. it's a local variable. needs a `var`